sim/src/events.rs

```diff
@@ -59,7 +59,8 @@ pub enum Event {
     },
     TripCancelled(TripID),
     TripPhaseStarting(TripID, PersonID, Option<PathRequest>, TripPhaseType),
-
+    TripIntersectionDelay(TripID, TurnID, u8),
```

Comment:
It's cheap to emit and process events; we only care about compressing to store in `Analytics`. I think it'd be more clear here to have `Duration` and do the compression to `u8` inside `analytics.rs`

(And later, I'll look into consolidating this and `IntersectionDelayMeasured`, which is just for traffic signals currently)

Reply:
That's fine, what about lane_speed_percentage?

Reply:
Changed

Reply:
Actually the only issue with this, is there is no way of having different thresholds for different transport modes (i.e walking) if we move it inside analytics. So is this something we are still happy about doing?

Reply:
You could also plumb `AgentID` in this event, then do `agent_id.to_type()` to change the threshold. It's cheap to send more info in events.